batch-models/src/main/scala/org/ekstep/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -174,8 +174,18 @@ object OnDemandDruidExhaustJob extends BaseReportsJob with Serializable with IJo
             // TO:DO
             // Fetch report config from dataset_metadata table
             val datasetConf = getDataSetDetails(requestType)
-            val reportConf = if(datasetConf.druid_query.nonEmpty) JSONUtils.deserialize[Map[String,AnyRef]](datasetConf.druid_query.get) else JSONUtils.deserialize[Map[String,AnyRef]](AppConf.getConfig("druid_query." + requestType))
+            var reportConf = if(datasetConf.druid_query.nonEmpty) JSONUtils.deserialize[Map[String,AnyRef]](datasetConf.druid_query.get) else JSONUtils.deserialize[Map[String,AnyRef]](AppConf.getConfig("druid_query." + requestType))
             val sortDfColNames = reportConf.get("sort").asInstanceOf[Option[List[String]]]
+            // Date Range with dynamic Start Date and End Date
+            val defaultInterval = reportConf.getOrElse("dateRange",None).asInstanceOf[Map[String,AnyRef]].getOrElse("interval",None).asInstanceOf[Map[String,AnyRef]]
```

Comment:
As said earlier also, Why can't we directly check on requestParamBody here, if its present replace else keep it as it is. Why so much of variables required here? Getting defaultStartDate from reportConf and appending it again back to reportConf if requestParamBody doesn’t have interval, instead do check on requestParamBody interval and alter reportConf only if it satisfies - which is simpler

Reply:
@sowmya-dixit As you requested, I have made the changes and pushed the code. Please check and merge the PR.

Thanks